### PR TITLE
added grammar to support multi-table refresh

### PR DESCRIPTION
**Pull request by mikethebeer**

### CODE REVIEW

File: sql-parser/src/main/java/io/crate/sql/tree/RefreshStatement.java [hunk 1]
```diff
@@ -24,21 +24,23 @@
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Objects;
 
+import java.util.List;
+
 public class RefreshStatement extends Statement {
 
-    private final Table table;
+    private final List<Table> tableList;
```
Comment (chaudum) [inline sql-parser/src/main/java/io/crate/sql/tree/RefreshStatement.java]: I would name it just `tables`

File: sql-parser/src/main/java/io/crate/sql/tree/RefreshStatement.java [hunk 2]
```diff
@@ -48,15 +50,23 @@ public boolean equals(Object o) {
...
 
-        if (!table.equals(that.table)) return false;
+        if (!tableList.equals(that.tableList)) return false;
 
         return true;
     }
 
     @Override
     public String toString() {
+        String tablenames = "";
```
Comment (chaudum) [inline sql-parser/src/main/java/io/crate/sql/tree/RefreshStatement.java]: does it need that?

File: sql/src/main/java/io/crate/action/sql/DDLStatementDispatcher.java [hunk 3]
```diff
@@ -315,27 +316,37 @@ public void onFailure(Throwable e) {
...
+        Map partitionNames = analysis.partitions();
+        final SettableFuture<Long> resultFuture = SettableFuture.create();
+        final AtomicInteger operations = new AtomicInteger(analysis.tables().size());
+
+        for (TableInfo tableInfo : analysis.tables()) {
+            String[] indexNames = getIndexNames(tableInfo, (PartitionName) partitionNames.get(tableInfo));
+            if (tableInfo.schemaInfo().systemSchema() || indexNames.length == 0) {
+                // shortcut when refreshing on system tables
+                // or empty partitioned tables
+                if (operations.decrementAndGet() == 0) { resultFuture.set(null); }
```
Comment (bogensberger) [inline sql/src/main/java/io/crate/action/sql/DDLStatementDispatcher.java]: add newlines

File: sql/src/main/java/io/crate/analyze/RefreshTableAnalyzedStatement.java [hunk 4]
```diff
@@ -28,28 +28,33 @@
...
-import java.util.Locale;
+import java.util.*;
 
 public class RefreshTableAnalyzedStatement extends AbstractDDLAnalyzedStatement {
 
     private final ReferenceInfos referenceInfos;
-    private TableInfo tableInfo;
-    private PartitionName partitionName;
+    private List<TableInfo> tableInfos;
+    private Map<TableInfo, PartitionName> partitions;
```
Comment (bogensberger) [inline sql/src/main/java/io/crate/analyze/RefreshTableAnalyzedStatement.java]: initialize directly and make them final if possible

**Comment by bogensberger** [comment]
otherwise ack

**Pull request merged**